src/services/permissions.service.ts

```diff
@@ -35,6 +40,33 @@ export const PermissionsService = (() => {
     }
   };
 
+  /**
+   * This function returns a boolean indicating whether the provided permission is present
+   * in the Permissions object.
+   *
+   * @param {any} permission A permission string to check.
+   * @returns {Boolean} True if the permission is valid, false otherwise.
+   */
+  const isValidPermission = (permission: unknown): boolean => {
+    if (typeof permission !== 'string') return false;
+
+    // Recursive function to navigate through the Permissions object without considering case.
+    const checkPath = (currentObject: any, remainingParts: string[]): boolean => {
+      if (remainingParts.length === 0) return true;
+
+      const currentPart = remainingParts.shift()!.toLowerCase();
+      if (currentPart === '*') return true; // Handle wildcard (*) permissions.
+      const matchingKey = Object.keys(currentObject).find((objKey) => currentPart === objKey.toLowerCase());
+      if (matchingKey) {
+        return checkPath(currentObject?.[matchingKey], remainingParts);
+      }
+
+      return false;
+    };
+
+    return checkPath(Permissions, permission.split('.'));
```

Comment:
This here works, and please correct me if I forgot a case I didn't cover but I believe we might be able to simplify this a little bit by:

1. Adding a new util `flattenObjectValues` in `src/utils/flattenObjectValues.util.ts` (+ unit tests)
  ```ts
  export const flattenObjectValues = (obj) =>
    Object.values(obj).flatMap(value =>
      value !== null && typeof value === "object"
        ? flattenObjectValues(value)
        : value
    );
```

2. Importing that util and using it here as such:
```suggestion
    const allPermissions = flattenObjectValues(Permissions)
    return allPermissions.includes(permission.toLowerCase())
```

Reply:
Good point. I approached it from the angle of navigating the object rather than flattening it. My thought was that this function will run several times and the permissions object might grow much larger but not much deeper. That approach would not scale with the size of Permissions. However this approach is clean and maintainable, and it's a pretty minuscule difference anyway.